packages-exp/auth-exp/src/core/persistence/persistence_user_manager.test.ts

```diff
@@ -0,0 +1,182 @@
+      const a = makePersistence();
+      const b = makePersistence();
+      const c = makePersistence();
+      const search = [a.persistence, b.persistence, c.persistence];
+      b.stub.get.returns(Promise.resolve(testUser('uid')));
+
+      const out = await PersistenceUserManager.create(mockAuth, search);
+      expect(out.persistence).to.eq(b.persistence);
+      expect(a.stub.get).to.have.been.calledOnce;
+      expect(b.stub.get).to.have.been.calledOnce;
+      expect(c.stub.get).not.to.have.been.called;
+    });
+
+    it('uses default user key if none provided', async () => {
+      const { stub, persistence } = makePersistence();
+      await PersistenceUserManager.create(mockAuth, [persistence]);
+      expect(stub.get).to.have.been.calledWith(
+        'firebase:authUser:test-api-key:test-app'
+      );
+    });
+
+    it('uses user key if provided', async () => {
+      const { stub, persistence } = makePersistence();
+      await PersistenceUserManager.create(
+        mockAuth,
+        [persistence],
+        'redirectUser'
+      );
+      expect(stub.get).to.have.been.calledWith(
+        'firebase:redirectUser:test-api-key:test-app'
+      );
+    });
+
+    it('returns zeroth persistence if all else fails', async () => {
+      const a = makePersistence();
+      const b = makePersistence();
+      const c = makePersistence();
+      const search = [a.persistence, b.persistence, c.persistence];
+      const out = await PersistenceUserManager.create(mockAuth, search);
+      expect(out.persistence).to.eq(a.persistence);
+      expect(a.stub.get).to.have.been.calledOnce;
+      expect(b.stub.get).to.have.been.calledOnce;
+      expect(c.stub.get).to.have.been.called;
+    });
+  });
+
+  describe('manager methods', () => {
+    let persistenceStub: sinon.SinonStubbedInstance<Persistence>;
+    let manager: PersistenceUserManager;
+
+    beforeEach(async () => {
+      const { persistence, stub } = makePersistence(PersistenceType.SESSION);
+      persistenceStub = stub;
+      manager = await PersistenceUserManager.create(mockAuth, [persistence]);
+    });
+
+    it('#setCurrentUser calls underlying persistence w/ key', async () => {
+      const user = testUser('uid');
+      await manager.setCurrentUser(user);
+      expect(persistenceStub.set).to.have.been.calledWith(
+        'firebase:authUser:test-api-key:test-app',
+        user
+      );
+    });
+
+    it('#removeCurrentUser calls underlying persistence', async () => {
+      await manager.removeCurrentUser();
+      expect(persistenceStub.remove).to.have.been.calledWith(
+        'firebase:authUser:test-api-key:test-app'
+      );
+    });
+
+    it('#getCurrentUser calls with instantiator', async () => {
+      const rawObject = {};
+      const userImplStub = sinon.stub(UserImpl, 'fromPlainObject');
+      persistenceStub.get.callsFake(
+        (_: string, cb?: (o: { [key: string]: unknown }) => any) => {
+          return cb!(rawObject);
```

Comment:
what's cb? why do we need to stub this other method?

Reply:
Done. Added a comment which should make this clear. The goal of this test is to ensure the instantiator is calling through to UserImpl.fromPlainObject